sdks/java/io/mongodb/src/main/java/org/apache/beam/sdk/io/mongodb/MongoDbIO.java

```diff
@@ -536,7 +545,7 @@ public List<BoundedSource<Document>> split(
             LOG.debug("using filters: " + allFilters.toJson());
             sources.add(new BoundedMongoDbSource(spec.withQueryFn(queryWithFilter)));
           }
-        } else {
+        } else if (spec.queryFn().getClass() == AutoValue_AggregationQuery.class) {
```

Comment:
Instead of throwing in split (which will fail the pipeline at runtime), it would be better to fail the pipeline during pipeline construction so that a user doesn't need to wait until the pipeline submits. This actually won't fail until the first time we try to split (which could take a bit), and I _think_ it won't fail the pipeline we'll just fail the split attempt (not 100% sure about that offhand).

We can do that by adding a check in expand like https://github.com/apache/beam/blob/f4204fbeb9daeb6c9261c3b424ed84870eb702ed/sdks/java/io/mongodb/src/main/java/org/apache/beam/sdk/io/mongodb/MongoDbIO.java#L334

Reply:
done, thanks